akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -810,8 +822,9 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
   }
 
   def createDecoder(compression: InboundCompressions, bufferPool: EnvelopeBufferPool): Flow[EnvelopeBuffer, InboundEnvelope, NotUsed] = {
+    val cache = new ActorRefResolveCache(provider, localAddress)
```

Comment:
This can't be shared across materializations. We currently don't do that, but perhaps we should stick to best practice. Either make `resolveActorRefWithLocalAddress` parameter a factory, or move the cache inside the `Decoder` stage and created there.